openapi/kas-fleet-manager.yaml

```diff
@@ -256,6 +256,87 @@ paths:
                   $ref: '#/components/examples/500Example'
     parameters:
       - $ref: "#/components/parameters/id"
+  /api/kafkas_mgmt/v1/kafkas/{id}/promote:
+    parameters:
+      - $ref: "#/components/parameters/id"
+      - in: query
+        name: async
+        description: Perform the action in an asynchronous manner. False by default.
+        schema:
+          type: boolean
+        required: true
+    post:
+      description: "Promote a Kafka instance. Promotion is performed asynchronously. The `async` query parameter has to be set to `true`. Only kafka instances with an `eval` billing_model are supported"
+      operationId: promoteKafka
+      requestBody:
+        description: "Kafka promotion request"
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/KafkaPromotionRequest'
+            examples:
+              KafkaPromotionRequestExample:
+                $ref: '#/components/examples/KafkaPromotionRequestExample'
+      responses:
+        "202":
+          # No 'content' attribute specified. This means no body is returned
+          description: Kafka promotion request accepted
+        "400":
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+              examples:
+                400CreationExample:
+                  $ref: '#/components/examples/400CreationExample'
+          description: Validation errors occurred
+        "401":
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+              examples:
+                401Example:
+                  $ref: '#/components/examples/401Example'
+          description: Auth token is invalid
+        "403":
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+              examples:
+                403Example:
+                  $ref: '#/components/examples/403Example'
+          description: User forbidden either because the user is not authorized to access the service.
+        "404":
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+              examples:
+                404Example:
+                  $ref: '#/components/examples/404Example'
+          description: The requested resource doesn't exist
+        "409":
```

Comment:
Will this error ever occur for this endpoint?

Reply:
I think at the moment no. Originally I added it in case we might need it